tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -331,8 +331,18 @@ public static boolean isTraceObservable() {
     static void setTrace(Trace trace) {
         currentTrace.set(trace);
 
-        // Give SLF4J appenders access to the trace id
+        // Give SLF4J appenders access to the trace id and if trace is being sampled
         MDC.put(Tracers.TRACE_ID_KEY, trace.getTraceId());
+        setTraceSampledMdcIfObservable(trace.isObservable());
+    }
+
+    private static void setTraceSampledMdcIfObservable(boolean isObservable) {
```

Comment:
nit: I don't like `is` in boolean names because it doesn't tell us anything that the type doesn't already make clear.